### PR TITLE
feat(unlock-app) - lock any lock for manage page

### PR DESCRIPTION
**Pull request by kalidiagne**
# Description

https://user-images.githubusercontent.com/20865711/199355739-8815dd88-2f13-4a77-9aa4-6b5651d4d17f.mov

Add  a way to handle a "Default" view where no lock is set in the query strings.

When the query string is not set, the user have to select network + lock 

Screenshot 2022-11-01 at 23 26 51

Screenshot 2022-11-01 at 23 26 57

Eventually the user can add a custom lock address, this can be done by clicking `Add custom address`, an input will show were the user can set the custom address, and after this have to confirm to see that specific lock.

**Custom lock** 
Screenshot 2022-11-01 at 23 37 58

Screenshot 2022-11-01 at 23 38 09

**Custom lock page**
Screenshot 2022-11-01 at 23 39 18

 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Review by julien51 — CHANGES_REQUESTED** [review]
Please make sure you replace all the places where the user has to select a lock with this!

File: unlock-app/src/components/interface/locks/Manage/elements/LockPicker.tsx [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+              {networkHasLocks ? (
+                <Select
+                  label="Lock"
+                  options={locksOptions}
+                  size="small"
+                  onChange={setLockAddress}
+                  defaultValue={defaultValues?.lockAddress}
+                />
+              ) : (
+                network && (
```
Comment (julien51) [inline unlock-app/src/components/interface/locks/Manage/elements/LockPicker.tsx:135]: We should always show the select, even when lock have been deployed with the option to ever another address...
Abd we should also make the link to deploy a new one visible at all times

Comment (kalidiagne) [reply unlock-app/src/components/interface/locks/Manage/elements/LockPicker.tsx]: **Other option**
Screenshot 2022-11-29 at 12 39 31

**Custom option**
Screenshot 2022-11-29 at 12 39 34

**Confirm custom option**
Screenshot 2022-11-29 at 12 39 48

Comment (julien51) [reply unlock-app/src/components/interface/locks/Manage/elements/LockPicker.tsx]: Great!

File: .clabot [hunk 2]
```diff
@@ -35,7 +35,9 @@
     "wonderwomancode",
     "devabdee",
     "Ether1oop",
-    "omahs"
+    "omahs",
+    "0xYee",
+    "SolomonFoskaay"
```
Comment (julien51) [inline .clabot:40]: Hum? Please don't change the `.clabot` config :D

File: packages/unlock-assets/package.json [hunk 3]
```diff
@@ -57,7 +57,7 @@
     "npm-run-all": "4.1.5",
     "prettier": "2.7.1",
     "react": "17.0.2",
-    "react-dom": "18.2.0",
+    "react-dom": "17.0.2",
```
Comment (julien51) [inline packages/unlock-assets/package.json:60]: Any reason for the downgrade?

File: .clabot (hunk 2 above)
Comment (kalidiagne) [reply .clabot:40]: mhh, let me check, i think there is something wrong with the rebase

**Comment by julien51** [comment]
Please add it to https://app.unlock-protocol.com/locks/settings as well!

**Comment by kalidiagne** [comment]
**Manage lock from settings page**
 
Screenshot 2022-11-30 at 19 35 32

Screenshot 2022-11-30 at 19 33 55

Screenshot 2022-11-30 at 19 31 12

**Comment by kalidiagne** [comment]
`Change lock` button removed 

Screenshot 2022-11-30 at 20 11 49

**Comment by julien51** [comment]
Perfect! LGTM

**Review by julien51 — APPROVED** [review]

**Pull request merged**